Library/Homebrew/formula_installer.rb

```diff
@@ -212,6 +213,10 @@ def install
       opoo "#{formula.full_name}: #{old_flag} was deprecated; using #{new_flag} instead!"
     end
 
+    invalid_flag_names.each do |flag|
+      opoo "#{formula.full_name}: #{flag} is invalid for this formula and will be ignored!"
```

Comment:
Sorry yes, I think see what `deprecated_option` does, but my idea here was to pick up typos - like `--with-postgreslq` and suggest `did you mean --with-postgresql?`. If the option given is not valid (or deprecated) and not close enough (e.g. Levenshtein - or whatever Homebrew uses) to a valid option it should be fatal yes - IMO. Could provide a continue & ignore option of course, if we go down UI route.